hibernate-core/src/main/java/org/hibernate/boot/models/xml/internal/XmlAnnotationHelper.java

```diff
@@ -1500,4 +1500,49 @@ public static void applyNotFound(
 		);
 		notFoundAnn.setAttributeValue( "action", notFoundAction );
 	}
+
+	public static void applySecondaryTables(List<JaxbSecondaryTableImpl> secondaryTables, MutableAnnotationTarget target, XmlDocumentContext xmlDocumentContext) {
+		if ( secondaryTables == null || secondaryTables.isEmpty() ) {
+			return;
+		}
+
+		final MutableAnnotationUsage<SecondaryTables> listenersUsage = target.replaceAnnotationUsage(
+				JpaAnnotations.SECONDARY_TABLE,
+				JpaAnnotations.SECONDARY_TABLES,
+				xmlDocumentContext.getModelBuildingContext()
+		);
+
+		final List<MutableAnnotationUsage<SecondaryTable>> values = arrayList( secondaryTables.size() );
+		listenersUsage.setAttributeValue( "value", values );
+
+		for ( JaxbSecondaryTableImpl secondaryTable : secondaryTables ) {
+			final MutableAnnotationUsage<SecondaryTable> tableAnn = JpaAnnotations.SECONDARY_TABLE
+					.createUsage( xmlDocumentContext.getModelBuildingContext() );
+			applyOr(
+					secondaryTable,
+					JaxbSecondaryTableImpl::getName,
+					"name",
+					tableAnn,
+					JpaAnnotations.SECONDARY_TABLE
+			);
```

Comment:
`name` is required, so this can just be

```suggestion
			tableAnn.setAttributeValue( "name", secondaryTable.getName() );
```